src/control/path_follower/path_follower/node_pure_pursuit.py

```diff
@@ -161,24 +169,42 @@ def path_callback(self, spline_path_msg: PathStamped):
 
             self.img_initialised = True
 
-    def callback(self, msg: PoseWithCovarianceStamped):
+    # def callback(self, msg: PoseWithCovarianceStamped):
+    def timer_callback(self):
```

Comment:
Thinking object oriented for the future (will make sense in the next PR):
 - timer callback code from here till line 208 could be one class function. `position` is then a class variable
 - another timer at the same rate run the the steering calcs depending on the method used: rvwp or attractive force. then calls a class function to return velocity with a passed steering angle.

then particle pursuit only has 1 function that replaces a pure pursuit function with no duplicate code.